### DISCUSSION TITLE
rename file in script with frn("spath" "dpath")

### DISCUSSION THREAD

**Discussion by DC-555** [arendst/Tasmota#19031] [Q&A] [answered]
I am not able to perform this function:
frn("spath" "dpath")

I put both just the file name and the whole path from the IP, but without success:
example:
(frn("/oldfile.txt", "/newfile.txt")

**Comment by sfromis** [comment]
What you have there looks odd, with that `(` at the start, and no matching `)`.
Not using scripting, but as I read the syntax, you should use blank as delimiter instead of `,`

**Comment by DC-555** [comment]
Yes, I tried this way too, and these others too:
frn("ould.txt" "new.txt")
frn("/ould.txt" "/new.txt")
frn("IP/ufs/ould.txt" "IP/ufs/new.txt")

**Comment by pkkrusty** [comment]
Try res=frn...

**Reply by DC-555** [reply]
This way it understands as a variable:
16:10:36.201 **var not found**: frn("slog.txt" "slognew.txt")

**Reply by gemu2015** [reply, marked as answer]
this is an extended command and needs this define
#define USE_SCRIPT_FATFS_EXT
correct syntax is this
frn("/ould.txt" "/new.txt")

**Reply by DC-555** [reply]
Thank you very much, it works perfectly.

**Reply by DC-555** [reply]
I have another doubt, I tried to change the name using a variable, because I would like to rename with increment, or date, is it possible?

**Reply by gemu2015** [reply]
of course it is
concatenate your file name string e.g.
fname="file_"+s(0hours)+"_"+s(0mins)
or
fname="file"+"_"+tstamp

**Reply by DC-555** [reply]
Thank you very much, gemu2015.
